lib/layout/data_grid.dart

```diff
@@ -366,8 +412,17 @@ class DataGridState extends WidgetState<DataGrid>
       if (w is! EnsembleDataRow) {
         throw Exception("Direct children of DataGrid must be of type DataRow");
       }
+
       EnsembleDataRow child = w;
-      if (!child.visible) {
+
+      // Callback to update the visible of a given row changes
+      child.onVisibleChanged = (value) {
+        setState(() {
+          rowVisibilities[index] = value;
```

Comment:
won't need this once you remove the optimization asI suggested above. You will still need setState but won't need to set the rowVisibilities